World builder3 add plugins

MFraters opened the pull request:
Follow up to #2571. This add the plugins which do the actual linking to the world builder.

gassmoeller commented on source/initial_composition/world_builder.cc:
```diff
@@ -0,0 +1,69 @@
+    {
+      return this->get_world_builder().composition(Utilities::convert_point_to_array(position),
+                                                   this->get_geometry_model().depth(position),
+                                                   n_comp) ? 1.0 : 0.0;
```
I do not understand the final `? 1.0 : 0.0`. Why does the world build not simply return the correct compositional value? E.g. what if you want to have a material that is 0.5  of two compositions at the same time?

MFraters replied:
hmm, I have changed my mind on this several times now. It is through this mechanism possible to have several compositions at the same time, but they all return true or false. I could change it back, so that it returns a double, but I have a hard time coming up with cases where you want to start a model with multiple compositions at the same location. Do you have something in mind?

gassmoeller replied:
Sorry for the late reply, was travelling. The prime example would be having grain size (or finite strain) in some compositional fields, and actual (chemical) compositions in the other fields. Also porosity in melt models is a separate field that is independent of the other fields.

gassmoeller commented on source/initial_composition/world_builder.cc:
```diff
@@ -0,0 +1,69 @@
+    ASPECT_REGISTER_INITIAL_COMPOSITION_MODEL(WorldBuilder,
+                                              "world builder",
+                                              "Specify the composition in through the World Buider located"
+                                              "in the ASPECT contrib directory. Make sure to specify the "
```
Same here, remove the reference to the directory. Do we already have a place with more documentation about the world builder? If so we should link to it from here. Also add to the second sentence the name of the world builder file parameter: `Make sure to specify the location of the World Builder file in the parameter ``World builder file'').

gassmoeller commented on source/initial_temperature/world_builder.cc:
```diff
@@ -0,0 +1,72 @@
+    ASPECT_REGISTER_INITIAL_TEMPERATURE_MODEL(WorldBuilder,
+                                              "world builder",
+                                              "Specify the initial temperature in through the World Buider located"
+                                              "in the ASPECT contrib directory. Make sure to specify the "
```
Same as for the composition.

bangerth commented:
The first three commits have already been merged. Can you rebase?

MFraters commented:
Sorry it took so long. I have been working focusing on first implementing on several crucial features like subduction zones and layers. Especially the implementation of suduction zones took a lot of time to figure out the right way of implementing it in spherical coordinates. I think that I have addressed all the comments, and it is ready for an other round of reviews.

MFraters commented:
Can anyone review this? It would be nice to have this merged soonish.

gassmoeller commented:
Looks good. Sorry for the delay.

MFraters commented:
Thanks!
